autorest/adal/token.go

```diff
@@ -247,6 +259,47 @@ func NewServicePrincipalTokenFromCertificate(oauthConfig OAuthConfig, clientID s
 	)
 }
 
+func NewServicePrincipalTokenFromMSI(oauthConfig OAuthConfig, resource string, callbacks ...TokenRefreshCallback) (*ServicePrincipalToken, error) {
+	// Read MSI settings
+	bytes, err := ioutil.ReadFile("/var/lib/waagent/ManagedIdentity-Settings")
```

Comment:
I have strong opinions against doing this. This location can absolutely *never* change or it will irreparably break *any application that uses this SDK w/ MSI*. The only way MSI works as an e2e feature is for the location to be completely permanent (or to have fallback locations in the future if it is moved).

Otherwise, what, now any downstream application configuration also has to include knowledge of where some latest version of the MSI extension is dropping its settings file in order to pass it into here? That just doesn't seem right.